DataSetExplorer/DataSetBuilder/Model/DataSetInstance.cs

```diff
@@ -55,11 +55,36 @@ internal bool HasNoAgreeingAnnotations()
 
         internal string GetSortedAnnotatorIds()
         {
-            var list = Annotations.Select(annotation => annotation.AnnotatorId).ToList();
+            var list = Annotations.Select(annotation => annotation.Annotator.Id).ToList();
             list.Sort();
             return string.Join(",", list);
         }
 
+        public int GetFinalAnnotation()
+        {
+            var majorityVote = GetMajorityAnnotation();
+            if (majorityVote != null) return (int)majorityVote;
+            return GetAnnotationFromMostExperiencedAnnotator();
+        }
+
+        private int? GetMajorityAnnotation()
+        {
+            var annotationsGroupedBySeverity = Annotations.GroupBy(a => a.Severity);
+            var severityCounts = annotationsGroupedBySeverity.Select(group => group.Count());
+            if (severityCounts.Any(count => count != severityCounts.First()))
```

Comment:
Intent of this statement is not clear - What is it checking? If there is more than one element in severityCounts that is not equal to first? How do we know the first element is the largest number?

Reply:
It does not have to be the largest, it is just important to check how many votes each severity got (e.g. two votes for severity 2 and two votes for severity 3). The goal of this condition is to check if all severities are equally voted - if they are not, then I take the severity that has the most votes.

Reply:
Got it, but it took me a while. I would extract the logic into a method, e.g.:

    if(HasMajoritySeverityVote(annotationsGroupedBySeverity)) { //includes the severityCounts creation.
        return annotationsGroupedBySeverity.Max(a => a.Count()).Key; //not sure if Max exists, so check.
    }

Reply:
Max gives the max number of votes for a severity, so I can't get the key (the actual severity).